sources/packages/api/src/route-controllers/institution-locations/institution-locations.controller.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from "@nestjs/common";
+import { InstitutionLocationService } from "../../services";
+import {
+  DesignationLocationAgreementStatus,
+  InstitutionLocationsDetailsDto,
+} from "./models/institution-location.dto";
+
+@Injectable()
+export class InstitutionLocationsControllerService {
+  constructor(private readonly locationService: InstitutionLocationService) {}
+
+  /**
+   * Retrieve institution locations and
+   * their associated designated status'.
+   * @param institutionId this value is passed only for client type Institution.
+   * @returns designation institution locations with their status'.
+   */
+  async getInstitutionLocationsWithDesignationStatus(
+    institutionId: number,
+  ): Promise<InstitutionLocationsDetailsDto[]> {
+    const institutionsWithDesignationStatus =
+      await this.locationService.getInstitutionLocationsWithDesignationStatus(
+        institutionId,
+      );
+    return institutionsWithDesignationStatus.map((el: any) => {
```

Comment:
Please avoid using any when the type is known.
I can see here that the properties are mapped like "location_id" because the way that they were retrieved using the getRawMany. We can avoid this doing the proper mapping after the getRawMany.

Reply:
yes, I agree with @andrewsignori-aot, and instead of any use can use `LocationWithDesigationStatus` and you can use alias. also, there are issues in mapping, 
![image](https://user-images.githubusercontent.com/77353155/160205879-cdc9c3c8-8ea0-4cdc-878b-f22b96038422.png)